src/Codeception/Application.php

```diff
@@ -0,0 +1,172 @@
+
+    /**
+     * Register commands from config file
+     *
+     *  extensions:
+     *      commands:
+     *          - project\MyNewCommand
+     *
+     */
+    public function registerCustomCommands()
+    {
+        try {
+            $this->readCustomCommandsFromConfig();
+        } catch (ConfigurationException $e) {
+            if ($e->getCode() === 404) {
+                return;
+            }
+            $this->renderException($e, new ConsoleOutput());
+            exit(1);
+        } catch (\Exception $e) {
+            $this->renderException($e, new ConsoleOutput());
+            exit(1);
+        }
+    }
+
+    /**
+     * Search custom commands and register them.
+     *
+     * @throws ConfigurationException
+     */
+    protected function readCustomCommandsFromConfig()
+    {
+        $this->getCoreArguments(); // Maybe load outside configfile
+
+        $config = Configuration::config();
+
+        if (empty($config['extensions']['commands'])) {
+            return;
+        }
+
+        foreach ($config['extensions']['commands'] as $commandClass) {
+            $commandName = $this->getCustomCommandName($commandClass);
+            $this->add(new $commandClass($commandName));
+        }
+    }
+
+    /**
+     * Validate and get the name of the command
+     *
+     * @param CustomCommandInterface $commandClass
+     *
+     * @throws ConfigurationException
+     *
+     * @return string
+     */
+    protected function getCustomCommandName($commandClass)
+    {
+        if (!class_exists($commandClass)) {
+            throw new ConfigurationException("Extension: Command class $commandClass not found");
+        }
+
+        $interfaces = class_implements($commandClass);
+
+        if (!in_array('Codeception\CustomCommandInterface', $interfaces)) {
+            throw new ConfigurationException("Extension: Command {$commandClass} must implement " .
+                                             "the interface `Codeception\\CustomCommandInterface`");
+        }
+
+        return $commandClass::getCommandName();
+    }
+
+    /**
+     * To cache Class ArgvInput
+     *
+     * @inheritDoc
+     */
+    public function run(InputInterface $input = null, OutputInterface $output = null)
+    {
+        if ($input === null) {
+            $input = $this->getCoreArguments();
+        }
+
+        return parent::run($input, $output);
+    }
+
+    /**
+     * Add global a --config option.
+     *
+     * @return InputDefinition
+     */
+    protected function getDefaultInputDefinition()
+    {
+        $inputDefinition = parent::getDefaultInputDefinition();
+        $inputDefinition->addOption(
+            new InputOption('config', 'c', InputOption::VALUE_OPTIONAL, 'Use custom path for config')
+        );
+        return $inputDefinition;
+    }
+
+    /**
+     * Search for --config Option and if found will be loaded
+     *
+     * example:
+     * -c file.yml
+     * --config file.yml
+     * --config=file.yml
```

Comment:
Two issues here:
1. -c is usually used to provide a path to directory, not a path to file,
example: php codecept build -c frameworks-yii-basic/tests
http://codeception.com/docs/reference/Commands 
https://github.com/Codeception/Codeception/blob/2.1/.travis.yml#L118
2. -cfile.yml (without spaces) must be supported too, to be fully compatible with http://symfony.com/doc/current/components/console/console_arguments.html